webpack/assets/javascripts/react_app/components/BookmarkForm/BookmarkForm.js

```diff
@@ -1,76 +1,199 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
-import * as Yup from 'yup';
+import { ExclamationCircleIcon } from '@patternfly/react-icons';
+import {
+  Button,
+  Form,
+  FormGroup,
+  TextInput,
+  Checkbox,
+  ActionGroup,
+  HelperText,
+  HelperTextItem,
+  FormHelperText,
+} from '@patternfly/react-core';
 import { useDispatch } from 'react-redux';
-import { noop } from '../../common/helpers';
-import ForemanForm from '../common/forms/ForemanForm';
-import TextField from '../common/forms/TextField';
-import { translate as __ } from '../../common/I18n';
-import { maxLengthMsg, requiredMsg } from '../common/forms/validators';
-import { submitForm } from '../../redux/actions/common/forms';
+import { sprintf, translate as __ } from '../../common/I18n';
+import { APIActions } from '../../redux/API';
+import './bookmarkForm.css';
 
 const BookmarkForm = ({
   url,
   controller,
-  onCancel,
   searchQuery,
   setModalClosed,
   bookmarks,
 }) => {
   const dispatch = useDispatch();
-  const existsNamesRegex = new RegExp(
-    `^(?!(${bookmarks.map(({ name }) => name).join('|')})$).+`
+  const existingNames = bookmarks.map(({ name }) => name);
+
+  const NAME_MAX_LENGTH = 255;
+  const SUCCESS_TYPE = 'success';
+  const ERROR_TYPE = 'error';
+
+  const nameTooLongMsg = sprintf(
+    __('name is too long, max %s'),
+    NAME_MAX_LENGTH
   );
-  const bookmarkFormSchema = Yup.object().shape({
-    name: Yup.string()
-      .max(...maxLengthMsg(254))
-      .required(requiredMsg())
-      .matches(existsNamesRegex, {
-        excludeEmptyString: true,
-        message: __('name already exists'),
-      }),
-    query: Yup.string()
-      .max(...maxLengthMsg(4096))
-      .required(requiredMsg()),
-  });
+  const nameExists = __('name already exists');
+  const blankInputErrMsg = __("Can't be blank");
+
+  const [name, setName] = useState('');
+  const [query, setQuery] = useState(searchQuery);
+  const [isPublic, setIsPublic] = useState(true);
+
+  const [nameErrorMsg, setNameErrorMsg] = useState('');
+  const [nameValidated, setNameValidated] = useState('');
+  const [queryErrorMsg, setQueryErrorMsg] = useState('');
+  const [queryValidated, setQueryValidated] = useState('');
+
+  const validateName = value => {
+    if (value.length > NAME_MAX_LENGTH) {
+      setNameValidated(ERROR_TYPE);
+      setNameErrorMsg(nameTooLongMsg);
```

Comment:
Should validate minimum length as well:
![Screenshot From 2025-07-03 18-25-48](https://github.com/user-attachments/assets/e5c274b5-ecee-4666-a1e6-b1799d3408e8)
![Screenshot From 2025-07-03 18-23-23](https://github.com/user-attachments/assets/1b36d1e9-aa30-4308-99a6-ca41d10bbd96)